packages/datasets-fs/src/commands/flatten.ts

```diff
@@ -0,0 +1,122 @@
+import { CaptchaItemTypes, Data, DataSchema, LabelledItem } from '@prosopo/types'
+import { InputOutputArgsSchema as InputOutputArgsSchema, InputOutputCliCommand } from '../utils/inputOutput.js'
+import { at, lodash } from '@prosopo/util'
+import { blake2b } from '@noble/hashes/blake2b'
+import { u8aToHex } from '@polkadot/util'
+import { z } from 'zod'
+import cliProgress from 'cli-progress'
+import fs from 'fs'
+
+export const ArgsSchema = InputOutputArgsSchema.extend({
+    allowDuplicates: z.boolean().optional(),
+})
+export type ArgsSchemaType = typeof ArgsSchema
+export type Args = z.infer<ArgsSchemaType>
+
+export class Flatten extends InputOutputCliCommand<ArgsSchemaType> {
+    public override getArgSchema() {
+        return ArgsSchema
+    }
+
+    public override getDescription(): string {
+        return 'Restructure a directory containing directories for each image classification into a single directory with a file containing the labels'
+    }
+
+    public override getOptions() {
+        return lodash().merge(super.getOptions(), {
+            input: {
+                description: 'Path to the data directory containing subdirectories for each image classification',
+            },
+            output: {
+                description: 'Where to put the output file containing the labels and single directory of images',
+            },
+            allowDuplicates: {
+                boolean: true,
+                description: 'If true, allow duplicates in the data',
+            },
+        })
+    }
+
+    public override async _run(args: Args) {
+        this.logger.debug('flatten run')
+        await super._run(args)
+
+        const dataDir = args.input
+        const outDir = args.output
+
+        // find the labels (these should be subdirectories of the data directory)
+        this.logger.info('reading data')
+        const labels: string[] = fs
+            .readdirSync(dataDir, { withFileTypes: true })
+            .filter((dirent) => dirent.isDirectory())
+            .map((dirent) => dirent.name)
+            .sort()
+        const imagesByLabel: string[][] = labels.map((label) => fs.readdirSync(`${dataDir}/${label}`))
+
+        // create the output directory
+        const imageDir = `${outDir}/images`
+        fs.mkdirSync(imageDir, { recursive: true })
+
+        const bar = new cliProgress.SingleBar({}, cliProgress.Presets.shades_classic)
+
+        // for each label
+        const items: LabelledItem[] = []
+        bar.start(
+            imagesByLabel.reduce((acc, images) => acc + images.length, 0),
+            0
+        )
+        for (let i = 0; i < labels.length; i++) {
+            // find all the images
+            const label = at(labels, i)
+            const images: string[] = at(imagesByLabel, i)
+            // for each image
+            for (const image of images) {
+                bar.increment()
+                // this.logger.log(`flattening ${label}/${image}`)
```

Comment:
Should this be commented out?

Reply:
yes because it messes with the progress bar that chris added. Unfortunately the progress bar lib cannot deal with printlns when it's active

Reply:
Can we remove instead of just commenting out?